emily/handler/src/database/accessors.rs

```diff
@@ -696,6 +696,66 @@ pub async fn set_api_state(context: &EmilyContext, api_state: &ApiStateEntry) ->
 
 // Limits ----------------------------------------------------------------------
 
+/// Returns height of oldest stacks block anchored to given bitcoin block
+async fn get_oldest_stacks_block_for_bitcoin_block(
+    context: &EmilyContext,
+    bitcoin_height: u64,
+) -> Result<u64, Error> {
+    query_with_partition_key::<ChainstateByBitcoinHeightTableSecondaryIndex>(
+        context,
+        &bitcoin_height,
+        None,
+        None,
+    )
+    .await?
+    .0
+    .iter()
+    .min_by_key(|entry| entry.key.height)
+    .map(|entry| entry.key.height)
+    .ok_or(Error::NotFound)
+}
+
+async fn calculate_sbtc_left_for_withdrawals(
+    context: &EmilyContext,
+    rolling_withdrawal_blocks: Option<u64>,
+    rolling_withdrawal_cap: Option<u64>,
```

Comment:
Actually, looks like these shouldn't be `Option`s at all, since we bail if they are `None`.

Reply:
Well, we should probably follow whatever is implemented in the signers since it will be less mental overhead. And I believe they treat these variables as a pair, where if they are both `None` then it means no limit. If one is `None` and not the other then that is an error state and it assumes the limit is zero.